### PR TITLE
Closes #953: Change csv generation from chart data

### LINKED ISSUE #953: Correct CSV export in analysis module
The CSV exporter should produce "column-wise" not "row-wise" data.
This would be **much more** useful, for further analysis in Excel,R, etc. 

Currently:

`1992; 1993; 1994`
`121; 122; 123`

After correction:
`1992; 121`
`1993; 122`
`1994; 123`

Perhaps some header would be also nice, say:

`Year;JudgmentCount`
`1992; 121`
`1993; 122`
`1994; 123`

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java [hunk 1]
```diff
@@ -1,83 +1,113 @@
...
  * A service for creating csv headers and rows
  * 
- * @author Łukasz Dumiszewski
+ * @author madryk
  */
 @Service("chartCsvGenerator")
 public class ChartCsvGenerator {
 
-    private PointValueFormatterManager pointValueFormatterManager;
+    private static final String COURT_NAME_HEADER = "Court";
```
Comment (lukdumi) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java]: Actually, it's not header but its part, the 1st column.

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java [hunk 2]
```diff
@@ -1,83 +1,113 @@
...
+     * Generates csv header for given {@link ChartCode} and {@link AnalysisForm}
      */
-    public String[] generateHeader(Chart<Object, Number> chart) {
+    public String[] generateHeader(ChartCode chartCode, AnalysisForm analysisForm) {
         
-        Preconditions.checkNotNull(chart);
+        Preconditions.checkNotNull(chartCode);
+        Preconditions.checkNotNull(analysisForm);
         
+        List<String> headers = Lists.newArrayList();
```
Comment (lukdumi) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java]: columnHeaders?

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java [hunk 3]
```diff
@@ -1,83 +1,113 @@
...
+            row.add(pointValueFormatterManager.format(series.getPoints().get(rowNumber).getY()));
+        }
+        
+        return row.toArray(new String[row.size()]);
+    }
+    
+    
+    //------------------------ PRIVATE --------------------------
+    
+    private String generateSeriesFilterHeader(AnalysisForm analysisForm, JudgmentSeriesFilter seriesFilter) {
```
Comment (lukdumi) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java]: generateColumnHeader?

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java [hunk 4]
```diff
@@ -1,83 +1,113 @@
...
+        if (StringUtils.isNotBlank(phrase)) {
+            headerName += " (" + phrase + ")";
         }
         
-        return row;
+        return headerName;
+        
+    }
+    
+    private String convertYValueTypetoHeaderName(UiyValueType valueType) {
```
Comment (lukdumi) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java]: convertYValue**T**ypeToHeaderName

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java [hunk 5]
```diff
@@ -1,83 +1,113 @@
...
         
+        switch(valueType) {
+            case NUMBER: return "JudgmentCount";
+            case NUMBER_PER_1000: return "JudgmentPer1000Judgments";
+            case PERCENT: return "JudgmentPercent";
```
Comment (lukdumi) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java]: Shouldn't it be lang dependent? Why in english?

Comment (madryk) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvGenerator.java]: I will add localization for this

**Pull request closed without merging**